modules/programs/zathura.nix

```diff
@@ -0,0 +1,53 @@
+{ config, lib, pkgs, ...}:
+
+with lib;
+
+let
+
+  cfg = config.programs.zathura;
+
+  formatLine = n: v:
+    let
+      formatValue = v:
+        if isBool v then (if v then "true" else "false")
+        else toString v;
+    in
+      "set ${n}\t\"${formatValue v}\"";
+in
+
+{
+  meta.maintainers = [maintainers.rprospero];
+
+  options.programs.zathura = {
+    enable = mkEnableOption ''Zathura is a highly customizable and funtional
+      document viewer focused on keyboard interaction.'';
+    options = mkOption {
+      default = null;
+      type = types.nullOr types.attrs;
+      description = ''
+      Add :set command options to zathura and make them permanent.
+      Run <code>man zathura</code> to see the full list of options
+      '';
+      example = literalExample ''
+        {default-bg = "#000000"; default-fg = "#FFFFFF";}
+      '';
+    };
+    extraConfig = mkOption {
+      type = types.lines;
+      default = "";
+      description = ''
+      Additional commands for zathura the zathurarc file.  If this and all
```

Comment:
The sentence
```xml
If this and all other zathura options are <code>null</code>, then this feature is disabled and no <filename>zathurarc</filename> link is produced.
```
doesn't seem true. From what I can tell the file will always be generated.

I'd suggest to keep producing the file and simply remove this sentence from the description. If anybody want to manage their own Zathura configuration then they could just add
```nix
home.packages = [ pkgs.zathura ];
```
to their configuration.

Reply:
Honestly, I was largely using the Xresources.nix file as a template for how to write this section.  I'd intended to put in that check early on, but came to the same conclusion you did.  I'm just embarassed that I left it in the comments.